drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (c) 2022 ITE Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_vcmp
+
+#include <device.h>
+#include <drivers/adc.h>
+#include <drivers/sensor.h>
+#include <drivers/sensor/it8xxx2_vcmp.h>
+#include <dt-bindings/dt-util.h>
+#include <dt-bindings/sensor/it8xxx2_vcmp.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(vcmp_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+#define VCMP_REG_MASK		0x7
+#define VCMP_RESOLUTION		BIT(10)
+#define VCMP_MAX_MVOLT		3000
+
+/* Device config */
+struct vcmp_it8xxx2_config {
+	/* Voltage comparator x control register */
+	uintptr_t reg_vcmpxctl;
+	/* Voltage comparator x channel select MSB register */
+	uintptr_t reg_vcmpxcselm;
+	/* Voltage comparator scan period register */
+	uintptr_t reg_vcmpscp;
+	/* Voltage comparator x threshold data buffer MSB register */
+	uintptr_t reg_vcmpxthrdatm;
+	/* Voltage comparator x threshold data buffer LSB register */
+	uintptr_t reg_vcmpxthrdatl;
+	/* Voltage comparator status register */
+	uintptr_t reg_vcmpsts;
+	/* Voltage comparator status 2 register */
+	uintptr_t reg_vcmpsts2;
+	/* Voltage comparator module irq */
+	int irq;
+	/* Voltage comparator channel */
+	int vcmp_ch;
+	/* Pointer of ADC device that will be performing measurement */
+	const struct device *adc;
+	/* Scan period for "all voltage comparator channel" */
+	int scan_period;
+	/*
+	 * Determines the condition between ADC data and threshold_mv
+	 * that will trigger voltage comparator interrupt.
+	 */
+	int comparison;
+	/* Threshold assert value in mv */
+	int threshold_mv;
+};
+
+/* Driver data */
+struct vcmp_it8xxx2_data {
+	/* ADC channel config */
+	struct adc_channel_cfg adc_ch_cfg;
+	/* Work queue to be notified when threshold assertion happens */
+	struct k_work work;
+	/* Sensor trigger hanlder to notify user of assetion */
+	sensor_trigger_handler_t handler;
+	/* Pointer of voltage comparator device */
+	const struct device *vcmp;
+};
+
+/* Voltage comparator work queue address */
+static uint32_t vcmp_work_addr[VCMP_CHANNEL_CNT] = {0};
+
+static void clear_vcmp_status(const struct device *dev, int vcmp_ch)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+	volatile uint8_t *reg_vcmpsts = (uint8_t *)config->reg_vcmpsts;
+	volatile uint8_t *reg_vcmpsts2 = (uint8_t *)config->reg_vcmpsts2;
+
+	/* W/C voltage comparator specific channel interrupt status */
+	if (vcmp_ch <= VCMP_CHANNEL_2)
+		*reg_vcmpsts = BIT(vcmp_ch);
+	else
+		*reg_vcmpsts2 = BIT(vcmp_ch - VCMP_CHANNEL_3);
```

Comment:
braces are required, please check coding guidelines

Reply:
Ah, I miss the brace here.